src/base/bittorrent/session.cpp

```diff
@@ -4129,6 +4132,11 @@ quint64 Session::getAlltimeUL() const
 
 void Session::refresh()
 {
+    if (!m_statsUpdateElapsedTimer.hasExpired(MINIMUM_STATS_UPDATE_INTERVAL_MSEC)) {
```

Comment:
the `Session` object technically already contains all the state it needs for this check. You could do this instead:

```
if (lt::clock_type::now() - m_statsLastTimestamp < MINIMUM_STATS_UPDATE_INTERVAL_MSEC) {
   return;
}
```

In fact, if you do, you could improve type-safety and improve readability by removing the `_MSEC` suffix on your constant and instead define it as:

```
const lt::clock_type::duration MINIMUM_STATS_UPDATE_INTERVAL = std::chrono::milliseconds(150); 
```

or:
```
const std::chrono::milliseconds MINIMUM_STATS_UPDATE_INTERVAL(150);
```

or perhaps even nicer:

```
using namespace std::chrono_literals;
const lt::clock_type::duration MINIMUM_STATS_UPDATE_INTERVAL = 150ms;
```

Either way, I don't think you need the `m_statsUpdateElapsedTimer`

Reply:
Thanks for the suggestion, but I personally don't like std::chrono or similar libraries, they are over-complicated, and I think in qbittorrent, people like to use Qt types more, but I will change it if anyone else disagrees.

Reply:
Hmm yeah I will try it

Reply:
right now, it's not clear that `hasExpired()` takes milliseconds. why not micro- or nanoseconds? or a `float` of seconds? I think it's unnecessary to leave readers with such uncertainty.